ruby/object_oriented_programming_basics/managing_ruby_projects.md

```diff
@@ -0,0 +1,227 @@
+
+Let's make sure whoever wants to use our script can do so:
+```bash
+bundle init
+bundle add colorize
+```
+This has created two files: `Gemfile` and `Gemfile.lock`. Let's take a look into both:
+```rb
+# Gemfile
+
+# frozen_string_literal: true
+
+source "https://rubygems.org"
+
+# gem "rails"
+
+gem "colorize", "~> 1.1"
+
+# Gemfile.lock
+GEM
+  remote: https://rubygems.org/
+  specs:
+    colorize (1.1.0)
+
+PLATFORMS
+  ruby
+  x86_64-linux
+
+DEPENDENCIES
+  colorize (~> 1.1)
+
+BUNDLED WITH
+   2.5.4
+```
+There's not much in those but as you can see, the `Gemfile` has information on where to get the gems from and what gems are required. The `"~> 1.1"` is something called a version constraint, particularly one called pessimistic constraint. It relies on semantic versioning - basically, the first number is the major version, second is the minor version and a third, if it exists is the patch number. Major versions can break things from previous versions - think changing method names for example. Minor versions can add and change things but they can't break anything. Patches happen when you introduce bug fixes that don't break anything. So, if people behind a gem maintain it in line with semantic versioning, you can rely on this pessimistic constraint never letting your project have a gem version that could potentially break your app - it is equivalent to `gem "colorize", ">= 1.1", "<2.0"`.
+
+`Gemfile.lock` has information on what was the last environment that should be able to run your app. Bundler will use it to install the same versions of gems even if `Gemfile` could potentially allow for newer versions to be installed.
+
+### .ruby-version
+
+There's another one small bit that is important to give to folks that will run your code: the target Ruby version of your project. You can do it easily by running `rbenv local 3.2.2` as it creates a `.ruby-version` file with the version declared. This is recognized by many other tools to figure out what Ruby version your project is running - for example, `rbenv` will no longer use the `global` Ruby version and the `Ruby LSP` VSCode extension will also change its behavior.
+
+### Ruby LSP in VSCode
+
+Earlier in the curriculum you've been instructed to choose the `No - launch without bundle` option when `Ruby LSP` told you about not finding a `Gemfile` - that also resulted with an error concerning `RuboCop`. In the next lesson we are going to go over `RuboCop` and how it should land in your project's `Gemfile`. After that, you are going to enjoy all the benefits of using `Ruby LSP` and its `RuboCop` integration and your projects are going to be set up like a real pro.
```

Comment:
Apologies, no, it doesn't but I want it to be part of this PR. I decided to ask for opinion already because I wasn't sure about the quality of the lesson. I'll add the modified bits of previous lessons soon TM.

Reply:
Added with [1dbff49](https://github.com/TheOdinProject/curriculum/pull/27128/commits/1dbff4924d758acbab2e6c08914ee75afe4cb5fb). I also took the liberty to remove the fluff about finding a different extension for different IDEs as the curriculum doesn't support anything but VSCode + it's rather self-explanatory that one should look for extensions for their IDE [bfbf12e](https://github.com/TheOdinProject/curriculum/pull/27128/commits/bfbf12ebff34774947ad597869434d2131d365bf).

Reply:
Had to update the wording here since in the meantime the message has changed which was what I used for `Installing Ruby` lesson but forgot to change it here.